singularity-eos/eos/eos_mgusup.hpp

```diff
@@ -125,6 +125,47 @@ class MGUsup : public EosBase<MGUsup> {
   FillEos(Real &rho, Real &temp, Real &energy, Real &press, Real &cv, Real &bmod,
           const unsigned long output,
           Indexer_t &&lambda = static_cast<Real *>(nullptr)) const;
+
+  PORTABLE_FORCEINLINE_FUNCTION
+  Real MinimumDensity() const { return 10 * robust::EPS(); }
```

Comment:
Since the reference isotherm scales as _rho0/rho, the EOS diverges at zero density. This bounds it at least to some degree.